archetype/engine-v2/src/test/resources/loader/versions.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2023 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<versions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+    <version>
+        <id>3.1.2</id>
```

Comment:
How about `<id>` -> `<archetype>` and `<supported-cli>` to `<cli>`.
A lot of archetype versions will be the same, so we probably should do multiples.

E.g. 
```xml
<version>
    <archetypes>
        <archetype>2.6.0</archetype>
        <archetype>2.6.1</archetype>
    </archetypes>
    <cli>[2.1,3)</cli>
</version>
```

The range is already encoded, we could also use a comma separated instead of multiple elements.

Reply:
Actually, here is a new revision on the file format:

```xml
<data>
    <versions>
        <version>2.0.0</version>
        <version>2.0.1</version>
        <version>2.0.2</version>
        <version>2.1.0</version>
        <version>2.2.0</version>
        <version>2.2.1</version>
        <version>2.2.2</version>
        <version>2.3.0</version>
        <version>2.3.1</version>
        <version>2.3.2</version>
        <version>2.3.3</version>
        <version>2.3.4</version>
        <version>2.4.0</version>
        <version>2.4.1</version>
        <version>2.4.2</version>
        <version>2.5.0</version>
        <version>2.5.1</version>
        <version>2.5.2</version>
        <version>2.5.3</version>
        <version>2.5.4</version>
        <version>2.5.5</version>
        <version>2.6.0</version>
        <version>3.0.0</version>
        <version>3.0.1</version>
        <version>3.0.2</version>
        <version>3.1.0</version>
        <version>3.1.1</version>
        <version>3.1.2</version>
    </versions>
    <rules>
        <!-- Helidon version 2.x archetypes are support by CLI 2.x, 3.x, 4.x -->
        <rule archetype="[2.0.0,3.0.0)" cli="[2.0.0,5.0.0)">
        <!-- Helidon version 3.x archetypes are support by CLI 3.x, 4.x -->
        <rule archetype="[3.0.0,4.0.0)" cli="[3.0.0,5.0.0)">
        <!-- Helidon version 4.x archetypes are support by CLI 4.x -->
        <rule archetype="[4.0.0,5.0.0)" cli="[4.0.0,5.0.0)">
    </rules>
</data>
```

You can implement the support for this file and omit `rules` for now.